tests/api/circulation/test_loan_bulk_extend.py

```diff
@@ -0,0 +1,176 @@
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Test bulk loan extension."""
+
+from datetime import timedelta
+
+import arrow
+from flask import current_app
+from invenio_circulation.api import Loan
+from invenio_circulation.pidstore.pids import CIRCULATION_LOAN_PID_TYPE
+from invenio_indexer.api import RecordIndexer
+from invenio_search import current_search
+
+from invenio_app_ils.circulation.api import bulk_extend_loans
+from invenio_app_ils.items.api import ITEM_PID_TYPE, Item
+from tests.api.conftest import _create_records
+from tests.helpers import user_login
+
+patron_username = 'patron1'
+pid_start_value = 99
+amount_of_extendable_loans = 4
+pid_end_value = pid_start_value + amount_of_extendable_loans
+date_format = "YYYY-MM-DD"
+item_loan_state = "ITEM_ON_LOAN"
+
+
+def create_loans(users, start=pid_start_value, end=pid_end_value):
+    extendable_loans = []
+
+    day_range_loan_extendable = \
+        current_app.config["ILS_CIRCULATION_LOAN_WILL_EXPIRE_DAYS"]
+    now = arrow.utcnow()
+    end_date_not_expired = now + timedelta(days=day_range_loan_extendable - 1)
+
+    patron_id = str(users[patron_username].id)
+
+    def loan_generator(pid_value):
+        loan = {
+            "document_pid": "docid-1",
+            "end_date": end_date_not_expired.format(date_format),
+            "item_pid": {
+                "type": "pitmid",
+                "value": f"itemid-{pid_value}"
+            },
+            "patron_pid": patron_id,
+            "pickup_location_pid": "locid-1",
+            "pid": f"loanid-{pid_value}",
+            "start_date": now.format(date_format),
+            "state": item_loan_state,
+            "transaction_location_pid": "locid-1",
+            "request_start_date": "2018-06-28",
+            "transaction_user_pid": patron_id,
+            "request_expire_date": "2019-06-28",
+            "patron": {
+                "email": "patron1@test.ch"
+            }
+        }
+
+        return loan
+
+    for number in range(start, end):
+        if len(extendable_loans) < amount_of_extendable_loans:
+            extendable_loans.append(loan_generator(number))
+
+    return extendable_loans
+
+
+def create_items(start=pid_start_value, end=pid_end_value):
+    items = []
+
+    def items_generator(pid_value):
+        item = {
+            "pid": f"itemid-{pid_value}",
+            "created_by": {"type": "script", "value": "demo"},
+            "barcode": f"123456{pid_value}89-1",
+            "document_pid": "docid-1",
+            "internal_location_pid": "ilocid-1",
+            "circulation_restriction": "NO_RESTRICTION",
+            "medium": "NOT_SPECIFIED",
+            "status": "CAN_CIRCULATE",
+            "document": {}
+        }
+        return item
+
+    for number in range(start, end):
+        items.append(items_generator(number))
+
+    return items
+
+
+def index_and_commit_data(db, users, repeated_run_number=0):
```

Comment:
This method is doing more that this, is also creating. Why not simply calling it:
```suggestion
def _prepare_data(db, users, number_of_loans=1):
```
?